linetools/analysis/zlimits.py

```diff
@@ -57,31 +59,35 @@ def from_dict(cls, idict):
         -------
 
         """
-        wrest = ltu.convert_quantity_in_dict(idict['wrest'])
-        slf = cls(wrest, idict['z'], idict['zlim'])
+        try:
+            wrest = ltu.convert_quantity_in_dict(idict['wrest'])
+        except KeyError:
+            wrest = None
+        slf = cls(idict['z'], idict['zlim'], wrest=wrest)
         # Return
         return slf
 
-    def __init__(self, wrest, z, zlim, **kwargs):
+    def __init__(self, z, zlim, wrest=None, **kwargs):
         """
         Parameters
         ----------
-        wrest : Quantity
-          Rest wavelength of the line.
         z : float
           Redshift of the line.
         zlim : tuple or list
           Redshift limits for a line
           Defined as wave/wrest - 1.
           Ok to have zlim[1]==zlim[0], but then self.is_set() == False
+        wrest : Quantity, optional
+          Rest wavelength.  Used for SpectralLine objects
```

Comment:
Used for SpectralLine objects -> Used for applying zLimits to individual SpectralLine objects

Reply:
edited